### PR TITLE
Remove scala.internal.quoted.LiftedExpr

### PR DESCRIPTION
**Pull request by nicolasstucki**
* Use reflection to implement `Liftable` for primitive types
* Add missing `Liftable[Byte]`
* Add `Expr.nullExpr` and `Expr.unitExpr` to lift those values without pickling

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/tastyreflect/KernelImpl.scala [hunk 1]
```diff
@@ -1454,6 +1480,9 @@ class KernelImpl(val rootContext: core.Contexts.Context, val rootPosition: util.
   def matchConstant_ClassTag(x: Constant): Option[Type] =
     if (x.tag == Constants.ClazzTag) Some(x.typeValue) else None
 
+  def Constant_apply(x: Unit | Null | Int | Boolean | Byte | Short | Int | Long | Float | Double | Char | String | Type): Constant =
+    Constants.Constant(x)
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/tastyreflect/KernelImpl.scala]: Why `Type`?

File: compiler/src/dotty/tools/dotc/transform/ReifyQuotes.scala [hunk 2]
```diff
@@ -199,8 +202,29 @@ class ReifyQuotes extends MacroTransform {
...
+          case _: Short => liftedValue(value, "Liftable_Short_delegate".toTermName, qctx)
+          case _: Int => liftedValue(value, "Liftable_Int_delegate".toTermName, qctx)
+          case _: Long => liftedValue(value, "Liftable_Long_delegate".toTermName, qctx)
+          case _: Float => liftedValue(value, "Liftable_Float_delegate".toTermName, qctx)
+          case _: Double => liftedValue(value, "Liftable_Double_delegate".toTermName, qctx)
+          case _: Char => liftedValue(value, "Liftable_Char_delegate".toTermName, qctx)
+          case _: String => liftedValue(value, "Liftable_String_delegate".toTermName, qctx)
+        }
+      }
+
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/transform/ReifyQuotes.scala]: We still need special compiler support for lifting primitives? Or it's for backward-compatibility.

File: library/src/scala/tasty/reflect/TypeOrBoundsOps.scala [hunk 3]
```diff
@@ -64,6 +64,8 @@ trait TypeOrBoundsOps extends Core {
 
   object Type {
 
+    def apply(clazz: Class[_])(implicit ctx: Context): Type = kernel.Type_apply(clazz)
+
     object IsConstantType {
```
Comment (liufengyun) [inline library/src/scala/tasty/reflect/TypeOrBoundsOps.scala]: Why the two neg-with-compiler tests are disabled?

File: compiler/src/dotty/tools/dotc/tastyreflect/KernelImpl.scala (hunk 1 above)
Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/tastyreflect/KernelImpl.scala]: That is the encoding for a `classOf[T]`

File: compiler/src/dotty/tools/dotc/transform/ReifyQuotes.scala (hunk 2 above)
Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/ReifyQuotes.scala]: We do not need it anymore.

**Pull request merged**